README.md

```diff
@@ -145,16 +178,17 @@ code, include a script or makefile that will do the job.
 
 ## A2.5
 
-«replace this with your answer»
-
+Wrote main.cpp up top. 
+run with :
+g++ main.cpp and then ./a.out
```

Comment:
Thank you for having tests, both positive and negative.

However, they didn't reveal a fairly large problem with the code. A sentence such as "the rabid foaming dog ate angrily" pass the tests. This is because you implemented the parser manually, rather than using a table as the question requested. Your code in (for example) state 1 should return false if neither of the `if` statements triggers, and instead just skips to the next word.

I was looking for something like

~~~ cpp
state_table states = {
  { 0, {{ "the", 1 }}},
	{ 1, {{ "lazy", 2}, {"smelly", 2}, {"dog", 3}, {"cat", 3}}},
	{ 2, {{ "lazy", 2}, {"smelly", 2}, {"dog", 3}, {"cat", 3}}},
	{ 3, {{ "ate", 4},  {"ran",    4}}},
	{ 4, {{ "slowly", 5}, { "noisily", 5}, {"EOI", END}}},
	{ 5, {{ "EOI", END }}}
};


int getNextState(int currentState, string nextWord) {
	transitions t = states[currentState];
	if (t.count(nextWord))
		return t[nextWord];
	else
		return -1;
}
~~

So -6 for not using a table, and -2 for the bugs:

# -8